extensions/amp-analytics/0.1/requests.js

```diff
@@ -166,22 +188,27 @@ export class RequestHandler {
    * @private
    */
   trigger_(isImmediate) {
+    if (this.queueSize_ == 0) {
+      // Do nothing if no request in queue
+      return;
+    }
+
     if (isImmediate) {
+      // If not batched, or batchInterval scheduler schedule trigger immediately
       this.fire_();
       return;
     }
 
-    // If is batched and not immediate
-    if (!this.timeoutId_) {
-      // schedule fire_ after certain time
-      this.timeoutId_ = this.win.setTimeout(() => {
+    // Schedule trigger after maxDelay.
+    if (this.maxDelay_ && !this.maxDelayTimeoutId_ && !this.batchInterval_) {
+      this.maxDelayTimeoutId_ = this.win.setTimeout(() => {
```

Comment:
Can't we overload maxDelay? maxDelay here is basically just one value from batchInterval. Just change the timeout set below to batchInterval[batchIntervalPointer].

Reply:
We talked about [this](https://github.com/ampproject/amphtml/issues/11241#issuecomment-330961343) before. With `maxDelay` we only start the timer after we trigger the first request and there's a greater possibility that a ping will be lost. `batchInterval` however flush batch queue independent of requests triggering. I believe what you want is `batchInterval`.